Assets/ModScripts/Submodules/UnscrewMaze.cs

```diff
@@ -0,0 +1,145 @@
+        Debug.LogFormat("[The Cruel Modkit #{0}] The center button is {1}.", ModuleID, Info.Arrows[8] == 9 ? "white. Use the arrow directions to navigate" : "grey. Use the arrow colors to navigate");
+        
+        curPos = positions[0];
+    }
+
+
+    public override void OnArrowPress(int Arrow)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.Audio.PlaySoundAtTransform(Module.ArrowSounds[Arrow].name, Module.transform);
+        Module.Arrows[Arrow].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} arrow button was pressed when the component selection was [{2}] instead of [{3}].", ModuleID, Info.ArrowDirections[Arrow], Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        if (Arrow > 3)
+            return;
+
+        Module.StartSolve();
+
+        int movementNum;
+        if (Info.Arrows[8] == 9)
+            movementNum = Arrow;
+        else
+        {
+            int[] movementIndices = { 2, 3, 1, 0 };
+            movementNum = Array.IndexOf(movementIndices, Info.Arrows[Arrow]);
+        }
+        if (!maze[curPos].Contains(movementNum.ToString()))
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! You hit a wall by moving {1} at the coordinates ({2}, {3}).", ModuleID, Info.ArrowDirections[movementNum].ToLower(), Math.Floor(curPos / 6f) + 1, (curPos % 6) + 1);
+            Module.CauseStrike();
+            return;
+        }
+        switch (movementNum)
+        {
+            case 0:
+                curPos -= 6;
+                break;
+            case 1:
+                curPos += 1;
+                break;
+            case 2:
+                curPos += 6;
+                break;
+            case 3:
+                curPos -= 1;
+                break;
+        }
+        UpdateMorse();
+    }
+
+    public override void OnBulbInteract(int Bulb)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.HandleBulbScrew(Bulb, BulbScrewedIn[Bulb], Info.BulbInfo[Bulb + 2]);
+
+        BulbScrewedIn[Bulb] = !BulbScrewedIn[Bulb];
+
+        Module.Audio.PlaySoundAtTransform(Module.BulbSounds[BulbScrewedIn[Bulb] ? 0 : 1].name, Module.transform);
+        Module.Bulbs[Bulb].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents() && !BulbScrewedIn[Bulb])
```

Comment:
```suggestion
        if (!Module.IsSolving() && !Module.CheckValidComponents()])
```
Check if module is already solving so that doesn't check components every time. Also use this log for screwing/unscrewing maybe?